grobid-core/src/main/java/org/grobid/core/data/BiblioItem.java

```diff
@@ -1727,132 +1728,132 @@ else if (string.startsWith("PACS Numbers") ||
-                bibtex += "@techreport{" + id + ",\n";
-            } else if (bookTitle != null) {
-                if ((bookTitle.startsWith("proc")) || (bookTitle.startsWith("Proc")) ||
-                        (bookTitle.startsWith("In Proc")) || (bookTitle.startsWith("In proc"))) {
-                    bibtex += "@inproceedings{" + id + ",\n";
-                } else {
-                    bibtex += "@article{" + id + ",\n"; // ???
-                }
+    /**
+     * Export to BibTeX format
+     *
+     * @param id                  the BibTeX ke to use
+     */
+    public String toBibTeX(String id, GrobidAnalysisConfig config) {
+        String type;
+        if (journal != null) {
+            type = "article";
+        } else if (book_type != null) {
+            type = "techreport";
+        } else if (bookTitle != null) {
+            if (StringUtils.containsIgnoreCase(bookTitle, "proceedings") ||
+                (bookTitle.startsWith("proc")) || (bookTitle.startsWith("Proc")) ||
+                (bookTitle.startsWith("In Proc")) || (bookTitle.startsWith("In proc"))) {
+                type = "inproceedings";
             } else {
-                bibtex += "@misc{" + id + ",\n"; // ???
+                LOGGER.debug("No journal given, but a booktitle. However, the booktitle does not start with \"proc\" or similar strings. Returning inbook");
+                type = "inbook";
             }
+        } else {
+            // using "misc" as fallback type
+            type = "misc";
+        }
+
+        StringJoiner bibtex = new StringJoiner(",\n", "@" + type + "{" + id + ",\n", "\n}\n");
+
+        try {
 
             // author 
             // fullAuthors has to be used instead
             if (collaboration != null) {
-                bibtex += "author\t=\t\"" + collaboration;
-            } else if (fullAuthors != null) {
-                if (fullAuthors.size() > 0) {
-                    boolean begin = true;
+                bibtex.add("  author = {" + collaboration + "}");
+            } else {
+                StringJoiner authors = new StringJoiner(" and ", "author = {", "}");
+                if (fullAuthors != null) {
                     for (Person person : fullAuthors) {
-                        if (begin) {
-                            bibtex += "author\t=\t\"" + person.getFirstName() + " " + person.getLastName();
-                            begin = false;
-                        } else
-                            bibtex += " and " + person.getFirstName() + " " + person.getLastName();
+                        authors.add(person.getLastName() + ", " + person.getFirstName());
                     }
-                    bibtex += "\"";
-                }
-            } else if (authors != null) {
-                StringTokenizer st = new StringTokenizer(authors, ";");
-                if (st.countTokens() > 1) {
-                    boolean begin = true;
+                } else {
```

Comment:
`this.authors` can be null (no author or no author extracted), we will get a NPE if we don't test it before. 

`else if (authors != null) {`

Reply:
OK. For the next "code iteration", I would like to try to introduce `Optional` in the model. ("Favor Optional Over Null" in [Java by Comparison](https://pragprog.com/book/javacomp/java-by-comparison))

Reply:
Yes this would be definitively an improvement! A lot of the java in Grobid has been developed long time ago, starting as a Java learning exercise... so it hurts me to look at it :D 

I have to say it's complicated to manage at the same time requests for new features, necessity for performance improvements, continuous redesign of ML, training data and code quality, given the low number of contributors in the project, but I guess this is the usual issue in Open Source projects which are a bit "niche".